diff --git a/web-ifc-three/src/IFC/components/subsets/SubsetCreator.ts b/web-ifc-three/src/IFC/components/subsets/SubsetCreator.ts
--- a/web-ifc-three/src/IFC/components/subsets/SubsetCreator.ts
+++ b/web-ifc-three/src/IFC/components/subsets/SubsetCreator.ts
@@ -60,6 +60,7 @@
     private filterIndices(config: SubsetConfig, subsetID: string) {
         const geometry = this.subsets[subsetID].mesh.geometry as IndexedGeometry;
         if (config.removePrevious) {
+            this.subsets[subsetID].ids.clear();
             geometry.setIndex([]);
             this.resetGroups(geometry);
             return;
