diff --git a/webapp/components/PostTeaser/PostTeaser.vue b/webapp/components/PostTeaser/PostTeaser.vue
--- a/webapp/components/PostTeaser/PostTeaser.vue
+++ b/webapp/components/PostTeaser/PostTeaser.vue
@@ -14,9 +14,6 @@
       <template v-if="post.image" #heroImage>
         <img :src="post.image | proxyApiUrl" class="image" />
       </template>
-      <client-only>
-        <user-teaser :user="post.author" :date-time="post.createdAt" />
-      </client-only>
       <h2 class="title hyphenate-text">{{ post.title }}</h2>
       <!-- TODO: replace editor content with tiptap render view -->
       <!-- eslint-disable vue/no-v-html -->
@@ -56,6 +53,9 @@
           />
         </client-only>
       </footer>
+      <client-only>
+        <user-teaser :user="post.author" :date-time="post.createdAt" />
+      </client-only>
     </base-card>
     <hc-ribbon
       :class="{ '--pinned': isPinned }"
diff --git a/webapp/pages/post/_id/_slug/index.vue b/webapp/pages/post/_id/_slug/index.vue
--- a/webapp/pages/post/_id/_slug/index.vue
+++ b/webapp/pages/post/_id/_slug/index.vue
@@ -22,30 +22,32 @@
           />
         </aside>
       </template>
-      <section class="menu">
-        <user-teaser :user="post.author" :date-time="post.createdAt">
-          <template #dateTime>
-            <ds-text v-if="post.createdAt !== post.updatedAt">({{ $t('post.edited') }})</ds-text>
-          </template>
-        </user-teaser>
-        <client-only>
-          <content-menu
-            placement="bottom-end"
-            resource-type="contribution"
-            :resource="post"
-            :modalsData="menuModalsData"
-            :is-owner="isAuthor"
-            @pinPost="pinPost"
-            @unpinPost="unpinPost"
-          />
-        </client-only>
-      </section>
-      <ds-space margin-bottom="small" />
       <h2 class="title hyphenate-text">{{ post.title }}</h2>
       <ds-space margin-bottom="small" />
       <content-viewer class="content hyphenate-text" :content="post.content" />
       <!-- eslint-enable vue/no-v-html -->
       <ds-space margin="xx-large" />
+      <!-- Shout Button -->
+      <div class="post-interaction">
+        <hc-shout-button
+          v-if="post.author"
+          :disabled="isAuthor"
+          :count="post.shoutedCount"
+          :is-shouted="post.shoutedByCurrentUser"
+          :post-id="post.id"
+        />
+      </div>
+      <ds-space margin-top="x-large">
+        <div class="post-interaction">
+          <hc-emotions :post="post" />
+        </div>
+      </ds-space>
+      <!-- Tags -->
+      <div v-if="post.tags && post.tags.length" class="tags">
+        <ds-space margin="xx-small" />
+        <hc-hashtag v-for="tag in sortedTags" :key="tag.id" :id="tag.id" />
+      </div>
+      <ds-space margin-bottom="small" />
       <!-- Categories -->
       <div class="categories">
         <ds-space margin="xx-small" />
@@ -61,32 +63,25 @@
           {{ post.language.toUpperCase() }}
         </ds-tag>
       </div>
+      <section class="menu">
+        <user-teaser :user="post.author" :date-time="post.createdAt">
+          <template #dateTime>
+            <ds-text v-if="post.createdAt !== post.updatedAt">({{ $t('post.edited') }})</ds-text>
+          </template>
+        </user-teaser>
+        <client-only>
+          <content-menu
+            placement="bottom-end"
+            resource-type="contribution"
+            :resource="post"
+            :modalsData="menuModalsData"
+            :is-owner="isAuthor"
+            @pinPost="pinPost"
+            @unpinPost="unpinPost"
+          />
+        </client-only>
+      </section>
       <ds-space margin-bottom="small" />
-      <!-- Tags -->
-      <div v-if="post.tags && post.tags.length" class="tags">
-        <ds-space margin="xx-small" />
-        <hc-hashtag v-for="tag in sortedTags" :key="tag.id" :id="tag.id" />
-      </div>
-      <ds-space margin-top="x-large">
-        <ds-flex :gutter="{ lg: 'small' }">
-          <ds-flex-item :width="{ lg: '75%', md: '75%', sm: '75%', base: '100%' }">
-            <hc-emotions :post="post" />
-          </ds-flex-item>
-          <!-- Shout Button -->
-          <ds-flex-item
-            :width="{ lg: '15%', md: '22%', sm: '22%', base: '100%' }"
-            class="shout-button"
-          >
-            <hc-shout-button
-              v-if="post.author"
-              :disabled="isAuthor"
-              :count="post.shoutedCount"
-              :is-shouted="post.shoutedByCurrentUser"
-              :post-id="post.id"
-            />
-          </ds-flex-item>
-        </ds-flex>
-      </ds-space>
       <!-- Comments -->
       <ds-section>
         <comment-list :post="post" @toggleNewCommentForm="toggleNewCommentForm" @reply="reply" />
@@ -293,6 +288,11 @@
     align-items: center;
   }
 
+  .post-interaction {
+    display: flex;
+    justify-content: center;
+  }
+
   &.--blur-image > .hero-image > .image {
     filter: blur($blur-radius);
   }
@@ -323,9 +323,4 @@
   }
 }
 
-@media only screen and (max-width: 960px) {
-  .shout-button {
-    float: left;
-  }
-}
 </style>
