diff --git a/viewer/src/components/ifc/ifc-manager.ts b/viewer/src/components/ifc/ifc-manager.ts
--- a/viewer/src/components/ifc/ifc-manager.ts
+++ b/viewer/src/components/ifc/ifc-manager.ts
@@ -50,7 +50,11 @@
    */
   async loadIfc(file: File, fitToFrame = false, onError?: (err: any) => any) {
     const url = URL.createObjectURL(file);
-    return this.loadIfcUrl(url, fitToFrame, undefined, onError);
+    try {
+      return await this.loadIfcUrl(url, fitToFrame, undefined, onError);
+    } finally {
+      URL.revokeObjectURL(url);
+    }
   }
 
   /**
