diff --git a/packages/table/src/table-body.js b/packages/table/src/table-body.js
--- a/packages/table/src/table-body.js
+++ b/packages/table/src/table-body.js
@@ -46,9 +46,17 @@
         </colgroup>
         <tbody>
           {
-            data.reduce((acc, row) => {
-              return acc.concat(this.wrappedRowRender(row, acc.length));
-            }, [])
+            data.reduce((acc, row) => {
+              const rendered = this.wrappedRowRender(row, acc.length);
+              if (Array.isArray(rendered)) {
+                for (let i = 0; i < rendered.length; i++) {
+                  acc.push(rendered[i]);
+                }
+              } else {
+                acc.push(rendered);
+              }
+              return acc;
+            }, [])
           }
           <el-tooltip effect={this.table.tooltipEffect} placement="top" ref="tooltip" content={this.tooltipContent}></el-tooltip>
         </tbody>
