diff --git a/src/service/plugin.service.ts b/src/service/plugin.service.ts
--- a/src/service/plugin.service.ts
+++ b/src/service/plugin.service.ts
@@ -16,11 +16,16 @@
   ) {
     const remoteList = ExtensionInformation.fromJSONList(remoteExt);
     const localList = this.CreateExtensionList();
+    const localFullNames = localList.map(ext =>
+      InstalledExtensionsService.GetExtensionFullName(ext)
+    );
 
     return remoteList.filter(
       ext =>
         !ignoredExtensions.includes(ext.name) &&
-        !localList.map(e => e.name).includes(ext.name)
+        !localFullNames.includes(
+          InstalledExtensionsService.GetExtensionFullName(ext)
+        )
     );
   }
 
@@ -29,6 +34,9 @@
     ignoredExtensions: string[]
   ) {
     const localExtensions = this.CreateExtensionList();
+    const remoteFullNames = remoteExtensions.map(ext =>
+      InstalledExtensionsService.GetExtensionFullName(ext)
+    );
 
     // for (var i = 0; i < remoteList.length; i++) {
 
@@ -51,7 +59,9 @@
     return localExtensions.filter(
       ext =>
         ext.name !== "code-settings-sync" &&
-        !remoteExtensions.map(e => e.name).includes(ext.name) &&
+        !remoteFullNames.includes(
+          InstalledExtensionsService.GetExtensionFullName(ext)
+        ) &&
         !ignoredExtensions.includes(ext.name)
     );
   }
