diff --git a/examples/intermediate/http-search-github/src/main.js b/examples/intermediate/http-search-github/src/main.js
--- a/examples/intermediate/http-search-github/src/main.js
+++ b/examples/intermediate/http-search-github/src/main.js
@@ -13,7 +13,7 @@
     .map(ev => ev.target.value)
     .filter(query => query.length > 0)
     .map(q => ({
-      url: `https://api.github.com/search/repositories?q=${encodeURI(q)}`,
+      url: `https://api.github.com/search/repositories?q=${encodeURIComponent(q)}`,
       category: 'github',
     }));
 
